design/object-store.md

```diff
@@ -0,0 +1,141 @@
+
+When the storage admin is ready to create an object storage, he will specify his desired configuration settings in a yaml file such as the following `object-store.yaml`. This example is a simple object store with metadata that is replicated across different hosts, and the data is erasure coded across multiple devices in the cluster.
+```yaml
+apiVersion: rook.io/v1alpha1
+kind: ObjectStore
+metadata:
+  name: my-store
+  namespace: rook
+spec:
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 3
+  dataPool:
+    failureDomain: device
+    erasureCoded:
+      dataChunks: 6
+      codingChunks: 2
+  gateway:
+    type: s3
+    port: 80
+    replicas: 3
+```
+
+Now create the object store.
+```bash
+kubectl create -f object-store.yaml
+```
+
+At this point the Rook operator recognizes that a new object store resource needs to be configured. The operator will create all of the resources to start the object store.
+1. Metadata pools are created (`.rgw.root`, `my-store.rgw.control`, `my-store.rgw.meta`, `my-store.rgw.log`, `my-store.rgw.buckets.index`)
+1. The data pool is created (`my-store.rgw.buckets.data`)
+1. A Ceph realm is created 
+1. A Ceph zone group is created in the new realm
+1. A Ceph zone is created in the new zone group
+1. A cephx key is created for the rgw daemon
+1. A Kubernetes service is created to provide load balancing for the RGW pod(s)
+1. A Kubernetes deployment is created to start the RGW pod(s) with the settings for the new zone
+
+When the RGW pods start, the object store is ready to receive the http or https requests as configured.
+
+
+## Object Store CRD
+
+The object store settings are exposed to Rook as a Custom Resource Definition (CRD). The CRD is the Kubernetes-native means by which the Rook operator can watch for new resources. The operator stays in a control loop to watch for a new object store, changes to an existing object store, or requests to delete an object store. 
+
+### Pools
+
+The pools are the backing data store for the object store and are created with specific names to be private to an object store. Pools can be configured with all of the settings that can be specified in the [Pool CRD](/Documentation/pool-crd.md). The underlying schema for pools defined by a pool CRD is the same as the schema under the `metadataPool` and `dataPool` elements of the object store CRD. All metadata pools are created with the same settings, while the data pool can be created with independent settings. The metadata pools must use replication, while the data pool can use replication or erasure coding. 
+
+```yaml
+  metadataPool:
+    failureDomain: host
+    replicated:
+      size: 3
+  dataPool:
+    failureDomain: device
+    erasureCoded:
+      dataChunks: 6
+      codingChunks: 2
+```
+
+### Gateway
+
+The gateway settings correspond to the RGW service. 
+- The `type` can be `S3` or `Swift`
+- If an ssl certificate is available, the RGW service will start with SSL termination enabled
```

Comment:
perhaps mention the format of the certificate, that it needs to be stored as a secret, etc.

Reply:
There is some overlap between this design doc and the user docs. I included that info in the user documentation in the other PR, but I'll mention it here too